src/targets/gpu/pack_int8_args.cpp

```diff
@@ -69,6 +93,76 @@ shape pack_int8_args::pack_int8_shape(const shape& s) const
     return {s.type(), lens, strides};
 }
 
+instruction_ref pack_int8_args::pad_ins(module& p, instruction_ref ins, int offset) const
+{
+    auto s                         = ins->get_shape();
+    auto lens                      = s.lens();
+    auto k                         = lens[lens.size() + offset];
+    auto pad_k                     = (k + 3) / 4 * 4;
+    auto pad_lens                  = lens;
+    pad_lens[lens.size() + offset] = pad_k;
+    std::vector<int64_t> pad_dims(lens.size() * 2, 0);
+    auto ret_ins = ins;
+    if(pad_k != k)
+    {
+        pad_dims[lens.size() + offset] = pad_k - k;
+        shape ps{s.type(), pad_lens};
+        auto ins_out = p.insert_instruction(ins, hip_allocate{ps});
+        op::pad pad{pad_dims};
+        ret_ins = p.insert_instruction(std::next(ins), hip_pad{pad}, ins, ins_out);
+    }
+
+    return ret_ins;
+}
+
+std::vector<instruction_ref> pack_int8_args::pad_inputs(module& p, instruction_ref ins) const
+{
+    std::vector<instruction_ref> ret_inputs;
+    auto inputs = ins->inputs();
+    auto in0    = inputs.at(0);
+    auto sa     = in0->get_shape();
+    bool transa = sa.transposed();
+    if(transa)
+    {
+        auto perm  = find_permutation(sa);
+        auto t_in  = in0->inputs().front();
+        int offset = static_cast<int>(perm.back()) - static_cast<int>(perm.size());
+        auto p_in  = pad_ins(p, t_in, offset);
+        auto val   = in0->get_operator().to_value();
+        assert(val.contains("dims"));
```

Comment:
This seems to assume that when the input is transposed there is a transpose operator, which is not a safe assumption.

Reply:
fixed